src/dto/PurchaseOptionsBuilder.ts

```diff
@@ -0,0 +1,100 @@
+import Product from "./Product";
+import {PurchaseUpdatePolicy} from "./enums";
+import ProductOfferDetails from "./storeProducts/ProductOfferDetails";
+import PurchaseOptions from "./PurchaseOptions";
+
+class PurchaseOptionsBuilder {
+    private offerId: string | null = null;
+    private applyOffer: boolean = true;
+    private oldProduct: Product | null = null;
+    private updatePolicy: PurchaseUpdatePolicy | null = null;
+    private contextKeys: string[] | null = null;
+    private quantity: number = 1;
+
+    /**
+     * iOS only.
+     * Set quantity of product purchasing. Use for consumable in-app products.
+     * @param quantity of product purchasing.
+     * @return builder instance for chain calls.
+     */
+    setQuantity(quantity: number): PurchaseOptionsBuilder {
+        this.quantity = quantity;
+        return this;
+    }
+
+    /**
+     * Set offer for the purchase.
+     * @param offer concrete offer which you'd like to purchase.
+     * @return builder instance for chain calls.
+     */
+    setOffer(offer: ProductOfferDetails) {
+        this.offerId = offer.offerId;
+        return this;
+    }
+
+    /**
+     * Set the offer Id to the purchase.
+     * If [offerId] is not specified, then the default offer will be applied. To know how we choose
+     * the default offer, see [QProductStoreDetails.defaultSubscriptionOfferDetails].
```

Comment:
It wouldn't work so. In rn we do smth like this
```suggestion
     * the default offer, see {@link QProductStoreDetails.defaultSubscriptionOfferDetails}.
```
double check pls